### ISSUE TITLE
Autosave bug in Firefox

### ISSUE THREAD

**Issue by jlau99999** [WordPress/gutenberg#10486]
Hello Gutenberg Developers,

When I open a pending post, I noticed autosave kicks in right away in Firefox.  If I try to publish the post while this is going on, the publishing job fails.  If I wait for the autosave job to complete then click publish, it works fine.  I do not see this behavior using Chrome.

Normally, I would just wait for the autosave job to complete then publish the content but I'm converting an existing plugin from classic editor to Gutenberg in which I am using .click() to invoke the publishing job and I don't know of a way to detect when autosave is running.  Is there a method to detect it?

I'm using the current version of Firefox and Gutenberg version 3.9.0 with Wordpress version 4.9.8.

Thanks,
-J

**Comment by mcsf** [comment]
@jlau99999, could you see if the issue is gone in [4.0 RC](https://make.wordpress.org/test/2018/10/10/call-for-testing-gutenberg-4-0-pre-release/)? Thanks!

**Comment by jlau99999** [comment]
Hi Miguel,

I tested with 4.0 RC and it's still happening in Firefox 64.0.3 64 bit.  

To reproduce, follow these steps.

1. Create a new post as an editor then submit for review.
2. Login as an administrator, go to the pending section then open the post.

Autosave will kick in as soon as the post is opened and that freezes everything for 10 to 20 seconds.  I couldn't unlock the post or publish until autosave is done.

The desired behavior would not have autosave start as soon as the post is opened.  This is a bug.

The workflow of autosave freezing out other user interactions is not desirable.  Clicking on Publish or taking over the post from another user should override autosave immediately rather than wait for 10 or 20 seconds for autosave to complete.  It took me a while to figure out the page was autosaving.  The typical user may not know what is happening.

Once again, this is only happening in Firefox.  I don't see any issue with the current version of Chrome.

Thanks,
-J

**Comment by noisysocks** [comment]
Hey @jlau99999! The autosave functionality has gone through a lot of changes this year. Would you mind confirming if this is still an issue when running the latest version of the Gutenberg plugin?

**Comment by skorasaurus** [comment]
following https://github.com/WordPress/gutenberg/issues/10486#issuecomment-429018718 
@jlau99999 's instructions,  I am unable to reproduce this behavior. 

When I open the post, within just a 1-2 seconds of clicking the edit button; the page is loaded and the 'saved' status appears next to the preview button in the upper right hand corner. 
![Selection_269](https://user-images.githubusercontent.com/955351/70443972-23a21580-1a67-11ea-8fcf-9cf787adb6d0.png)

Tested on firefox 70.0.1; gutenberg 7.0.0. no other plugins were enabled. used the twentytwenty theme. 
Note, that I did test this on a local testing site on a vvv environment, I don't know if that has anything to do with it.
